commit 50bc58fee203207196bcf7974a5dfcf0cc6fa67d (refs/changes/75/4975/4)
Author: Rui Loura <rui@joyent.com>
Date:   2018-11-20T20:20:24+00:00 (11 months ago)
    
    TRITON-841 Add “Admin IP” and “Admin_IP” to sysinfo output

diff --git a/src/sysinfo b/src/sysinfo
index 8f1a3f72..3d833443 100644
--- a/src/sysinfo
+++ b/src/sysinfo
@@ -539,13 +539,21 @@ function get_overlay_nic_tags()
     fi
 }
 
+# Default to Admin_NIC_Tag of "admin"
 function get_admin_nic_tag()
 {
     prop=$(net_conf_grep "^admin_tag=")
     fields=(${prop//=/ })
-    eval "Admin_NIC_Tag=${fields[1]}"
+    eval "Admin_NIC_Tag=${fields[1]:-"admin"}"
 }
 
+function get_admin_ip()
+{
+    tag=${Admin_NIC_Tag:-"admin"}
+    prop=$(net_conf_grep "^${tag}_ip=")
+    fields=(${prop//=/ })
+    eval "Admin_IP=${fields[1]}"
+}
 
 function get_nic_tags()
 {
@@ -696,6 +704,7 @@ Bhyve_Max_Vcpus=${Bhyve_Max_Vcpus}
 HVM_API='${HVM_API}'
 Nic_Tags=${NicTagList}
 Admin_NIC_Tag=${Admin_NIC_Tag}
+Admin_IP=${Admin_IP}
 Setup='${Setup_complete}'
 END
     else
@@ -834,6 +843,7 @@ END
   "CPU Virtualization": "${CPU_Virtualization}",
   "CPU Physical Cores": ${CPU_Count},
   "Admin NIC Tag": "${Admin_NIC_Tag}",
+  "Admin IP": "${Admin_IP}",
 END
     else
         echo "  \"ZFS Quota\": \"${QUOTA}\","
@@ -1025,6 +1035,7 @@ get_system_type
 get_hostname
 get_aggregation_mappings
 get_overlay_nic_tags
+get_admin_ip
 get_admin_nic_tag
 get_nic_tags
 get_smartos_network_interfaces
